control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -535,6 +587,111 @@ func (r *HostedControlPlaneReconciler) ensureControlPlane(ctx context.Context, h
 	return nil
 }
 
+func (r *HostedControlPlaneReconciler) reconcileEtcd(ctx context.Context, hcp *hyperv1.HostedControlPlane, releaseImage *releaseinfo.ReleaseImage) error {
+	rootCASecret := manifests.RootCASecret(hcp.Namespace)
+	if err := r.Get(ctx, client.ObjectKeyFromObject(rootCASecret), rootCASecret); err != nil {
+		return fmt.Errorf("cannot get root CA secret: %w", err)
+	}
+
+	// Etcd client secret
+	clientSecret := manifests.EtcdClientSecret(hcp.Namespace)
+	if err := r.Get(ctx, client.ObjectKeyFromObject(clientSecret), clientSecret); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("cannot get client secret: %w", err)
+	}
+	if _, err := controllerutil.CreateOrUpdate(ctx, r, clientSecret, func() error {
+		clientSecret.OwnerReferences = ensureHCPOwnerRef(hcp, clientSecret.OwnerReferences)
+		return ReconcileEtcdClientSecret(clientSecret, rootCASecret)
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile etcd client secret: %w", err)
+	}
+
+	// Etcd server secret
+	serverSecret := manifests.EtcdServerSecret(hcp.Namespace)
+	if err := r.Get(ctx, client.ObjectKeyFromObject(clientSecret), clientSecret); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
why do we need this Get request here? what is it giving us that is not already handled by CreateOrUpdate?